pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSchemaRestletResource.java

```diff
@@ -215,22 +240,19 @@ private SuccessResponse addOrUpdateSchema(@Nullable String schemaName, Schema sc
     }
 
     try {
-      SchemaEventType eventType = SchemaEventType.UPDATE;
-      if (schemaName == null) {
-        // New schema posted
-        eventType = SchemaEventType.CREATE;
-      }
-      _pinotHelixResourceManager.addOrUpdateSchema(schema);
-
+      _pinotHelixResourceManager.updateSchema(schema, reload);
       // Best effort notification. If controller fails at this point, no notification is given.
-      LOGGER.info("Metadata change notification for schema {}, type {}", schema, eventType);
-      _metadataEventNotifierFactory.create().notifyOnSchemaEvents(schema, eventType);
-
+      LOGGER.info("Notifying metadata event for updating schema: {}", schemaName);
+      _metadataEventNotifierFactory.create().notifyOnSchemaEvents(schema, SchemaEventType.UPDATE);
       return new SuccessResponse(schema.getSchemaName() + " successfully added");
+    } catch (SchemaNotFoundException | TableNotFoundException e) {
```

Comment:
Can we handle these 2 exceptions separately and return different error message?
For SchemaNotFoundException: "Failed to find schema %s"
For TableNotFoundException: "Failed to find table %s to reload"